django_mongodb/query.py

```diff
@@ -120,6 +133,75 @@ def get_cursor(self):
         return self.collection.aggregate(pipeline)
 
 
+def join(self, compiler, connection):
+    lookups_pipeline = []
+    join_fields = self.join_fields or self.join_cols
+    lhs_fields = []
+    rhs_fields = []
+    for lhs, rhs in join_fields:
+        if isinstance(lhs, str):
```

Comment:
What case is this? I haven't been able to find the tests that use this branch.

Reply:
I imported the logic from https://github.com/WaVEV/django/blob/main/django/db/models/sql/datastructures.py#L88
In that join function there is a second part of the join, but I didn't find where it is used.

Reply:
Thanks. So it's to support [a deprecated private API](
https://github.com/django/django/commit/8b1ff0da4b162e87edebd94e61f2cd153e9e159d). I think we should just simplify the code now and not worry about this. It's untested by Django's test suite anyway (checked running with PostgreSQL). Any third-party apps should be on their way to supporting the new API by now.